### PR TITLE
docs(expect): document methods and add examples

### PR DESCRIPTION
**Pull request by satyarohith**
- **add docs for fn module**
- **add docs for expect module**

### CODE REVIEW

File: expect/expect.ts [hunk 1]
```diff
@@ -100,108 +103,317 @@ const matchers: Record<MatcherKey, Matcher> = {
...
               applyMatcher(value, args)
             )
-            : applyMatcher(value, args);
-        };
-      },
+          : applyMatcher(value, args);
+      };
     },
-  );
+  });
```
Comment (satyarohith) [inline expect/expect.ts:219]: Note: just format diff

**Review by iuioiua — CHANGES_REQUESTED** [review]
Can you please add `../expect/mod.ts` to the [doc checker](https://github.com/denoland/deno_std/blob/83991c738edc33dcc63a099f505781a6f01809df/_tools/check_docs.ts#L28)?

File: expect/expect.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
 // Copyright 2019 Allain Lalonde. All rights reserved. ISC License.
+// Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (kt3k) [inline expect/expect.ts:4]: 👍

File: expect/expect.ts [hunk 3]
```diff
@@ -100,108 +107,348 @@ const matchers: Record<MatcherKey, Matcher> = {
...
-expect.any = asymmetricMatchers.any;
-expect.arrayContaining = asymmetricMatchers.arrayContaining;
-expect.closeTo = asymmetricMatchers.closeTo;
-expect.stringContaining = asymmetricMatchers.stringContaining;
-expect.stringMatching = asymmetricMatchers.stringMatching;
+/**
+ * This is a no-op right now. Don't use it.
+ * Todo: support serialize plugin
+ *
+ * @ts-
```
Comment (kt3k) [inline expect/expect.ts:304]: remove?
```suggestion
```

**Review by kt3k — APPROVED** [review]
LGTM. Thanks, Satya!

**Pull request merged**